### PR TITLE
[SRTS-A01](Aqara Smart Radiator Thermostat E1) Added smart schedule support

### PR DESCRIPTION
**Pull request by artem-sedykh**

### CODE REVIEW

**Comment by artem-sedykh** [comment]
image
image

File: devices/xiaomi.js [hunk 1]
```diff
@@ -211,16 +242,84 @@ const tzLocal = {
...
+                break;
+            }
             default: // Unknown key
                 meta.logger.warn(`zigbee-herdsman-converters:aqara_trv: Unhandled key ${key}`);
             }
         },
         convertGet: async (entity, key, meta) => {
-            const dict = {'system_mode': 0x0271, 'preset': 0x0272, 'window_detection': 0x0273, 'valve_detection': 0x0274,
-                'child_lock': 0x0277, 'away_preset_temperature': 0x0279, 'calibrated': 0x027b, 'sensor': 0x027e};
+            const dict = {'state': [0x0271], 'preset': [0x0272], 'window_detection': [0x0273], 'valve_detection': [0x0274],
```
Comment (trullock) [inline devices/xiaomi.js:321]: Shouldnt 'state' here be 'system_mode', see the last merged PR. Think youve conflicted here

**Comment by artem-sedykh** [comment]
**Start and end time duration at least 4 hours**  

image

**The end time must be greater than the start time, minimum duration is one hour**
image

File: devices/xiaomi.js [hunk 2]
```diff
@@ -211,16 +242,88 @@ const tzLocal = {
...
+            case 'schedule_options': {
+                if (value.intervals.length != 3) {
+                    throw new Error('the number of intervals must be equal to four');
+                }
+
+                if (value.repeat.length == 0) {
+                    throw new Error('"repeat" must have at least one value');
+                }
+
+                const dayNames = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
```
Comment (Koenkk) [inline devices/xiaomi.js:258]: Please make the day names lowercase

File: devices/xiaomi.js [hunk 3]
```diff
@@ -2394,6 +2497,8 @@ module.exports = [
             exposes.binary('window_open', ea.STATE, true, false),
             e.valve_detection().setAccess('state', ea.ALL),
             e.away_preset_temperature().withAccess(ea.ALL),
+            exposes.switch()
+                .withState('schedule', true, 'Smart schedule', ea.ALL, 'ON', 'OFF'),
```
Comment (Koenkk) [inline devices/xiaomi.js:2501]: Can yo ualso add a description here? (use `.withDescription('blabla')`)

File: devices/xiaomi.js [hunk 4]
```diff
@@ -2394,6 +2497,8 @@ module.exports = [
             exposes.binary('window_open', ea.STATE, true, false),
             e.valve_detection().setAccess('state', ea.ALL),
```
Comment (Koenkk) [inline devices/xiaomi.js:2498]: I see `schedule_options` was not added to exposes, can you  add that? Looks like you will need to use the `composite` exposes.

**Comment by problem9** [comment]
is this working, will be this merged?

**Comment by protyposis** [comment]
What is missing to get this merged? What is the general PR review process in this repo?

**Comment by Koenkk** [comment]
@artem-sedykh could you add the `schedule_settings` to the exposes?

**Comment by artem-sedykh** [comment]
I don't know how to do it, now it works via mqtt

**Comment by Koenkk** [comment]
@artem-sedykh can you provide me an example on what payload `schedule_setting` has?

**Review by protyposis — COMMENTED** [review]
I am wondering whether the Z2M Frontend supports JSON data, or if the `schedule_settings` should be additionally stringified to allow modification in the UI.

**Comment by 0ip** [comment]
Not much is happening here anymore, unfortunately, but I think the feature is relatively important to many owners of that thermostat, me included.

@protyposis Would it be ok if you open a new pull request based on the excellent work of @artem-sedykh and your suggestions?

File: devices/xiaomi.js [hunk 5]
```diff
@@ -82,12 +82,46 @@ const fzLocal = {
...
                     result['valve_alarm'] = {1: true, 0: false}[value];
                     break;
+                case 0x00f7:
+                    result['valve_alarm'] ={1: true, 0: false}[value.readUInt32LE(36)];
+                    result['occupied_heating_setpoint'] = value.readUint16LE(32)/100;
+                    result['local_temperature'] = value.readUint16LE(28)/100;
+                    break;
+                case 0x027d:
+                    result['schedule'] = {1: 'ON', 0: 'OFF'}[value];
+                    break;
```
Comment (protyposis) [inline devices/xiaomi.js:92]: How does this relate to the `preset` setting? Shouldn't the schedule be active when `preset` is set to `auto`, or does `auto` mean something different? What happens when I set `preset` to `manual` or `away`, and at the same time `schedule` to `ON`?

**Comment by protyposis** [comment]
@0ip I'd love to but I'm afraid I'm missing the background to be able to decide what makes sense here. The only knowledge I have of Zigbee messages in general and the Aqara TRV specifically is what I read in this PR, and that unfortunately seems insufficient at this point.

**Comment by Koenkk** [comment]
@0ip feel free to make a new pr

**Comment by 0ip** [comment]
@Koenkk I find myself in a similar situation as @protyposis, sorry

**Comment by protyposis** [comment]
Gave it a try anyway: https://github.com/Koenkk/zigbee-herdsman-converters/pull/5058, @Koenkk if you could be so kind and review please :)

**Comment by Koenkk** [comment]
I believe this one can be closed in favour of https://github.com/Koenkk/zigbee-herdsman-converters/pull/5058

**Pull request closed without merging**